src/application.py

```diff
@@ -0,0 +1,84 @@
+import json
+from dataclasses import dataclass
+from json import JSONEncoder
+from typing import List
+from wsgiref import simple_server
+
+import falcon
+
+import arena
+import database
+import dnd5eapi
+
+
+class ArenaResource:
+    def on_post(self, req, resp):
+        fight_request = self.FightRequest(**req.media)
+        if fight_request.max_rounds <= 0:
+            raise AttributeError(f"max_rounds: {fight_request.max_rounds} <=0!")
+        if len(fight_request.combatants) < 2:
+            raise AttributeError(
+                f"not enough combatants: {len(fight_request.combatants)}"
+            )
+
+        battle_result: arena.BattleResult = arena.Arena(
+            self.read_combatant(fight_request.combatants[0]),
+            self.read_combatant(fight_request.combatants[1]),
+            fight_request.max_rounds,
+        ).fight()
+        database.write_battle_result(battle_result)
+        resp.status = falcon.HTTP_200
+        resp.body = json.dumps(battle_result, cls=self.BattleResultEncoder)
```

Comment:
Wouldn't it be better to just write here `battle_result.__dict__` or `dataclasses.asdict(battle_result)`?